app/client/src/contexts/Fullscreen.tsx

```diff
@@ -0,0 +1,37 @@
+import { createContext, useContext, useMemo, useState } from 'react';
+import type { ReactNode } from 'react';
+
+type State = {
+  fullscreenActive: boolean;
+  setFullscreenActive: (fullscreenActive: boolean) => void;
+};
+
+// --- components ---
+const FullscreenContext = createContext<State | undefined>(undefined);
+
+type Props = {
+  children: ReactNode;
+};
+
+export function FullscreenProvider({ children }: Props) {
+  const [fullscreenActive, setFullscreenActive] = useState(false);
+  const state: State = useMemo(() => {
+    return { fullscreenActive, setFullscreenActive };
+  }, [fullscreenActive]);
+
+  return (
+    <FullscreenContext.Provider value={state}>
+      {children}
+    </FullscreenContext.Provider>
+  );
+}
+
+export function useFullscreenContext() {
```

Comment:
This was a recommendation from @courtneymyers: it's a wrapper to ensure the context is being accessed from inside a provider.

Reply:
Makes sense. 

I guess I'll flip the question. Should the `AddDataWidget` context follow the pattern of the `Fullscreen` and `MapHighlight` contexts?

Reply:
Thanks, Max. @cschwinderg it saves you from having to use undefined checks throughout your code anytime you want to access what's stored in that context component by moving that check into the custom hook. I learned that from a [Kent C. Dodds blog post](https://kentcdodds.com/blog/how-to-use-react-context-effectively#typescript) in case you want to read more about that pattern.

@maxdiebold-erg one update – I'd recommend calling the custom hook `useFullscreenState()` to be explicit what that context component holds (state in this case). I do that to differentiate from similar custom hooks I create that return the `dispatch` function stored in a separate context component (see https://github.com/Eastern-Research-Group/mywaterway/blob/develop/app/client/src/contexts/FetchedData.tsx#L243-L265 where I'm doing that).

In this case, you're using `useState` instead of `useReducer` so no need for a second context component and it's accompanying custom hook, but I like to be consistent across the app.

Reply:
@cschwinderg Yes, we should probably do the same for the `AddDataWidget` context, too, I'll switch that over.

@courtneymyers sounds good, I'll do the same for the others as well.